### ISSUE TITLE
Speed up when users only submit a few points

### ISSUE THREAD

**Issue by kdaust** [bcgov/climr#126]
Currently, climr checks the bounding box of the points, and downloads the data for the box. In general, I think this works well, because with caching, data is often reusable. However, in the case where only a few (< 500) points over a large spatial area, it may not make sense to need to download the full dataset. A couple options:

1. Send the points to the database and extract the data there directly. This theoretically is nice, but the bilinear extraction is quite slow - 350 points take about 50 seconds to process (better than ~5 mins for downloading the whole dataset, but still not great). Of course if we're down to ~50 points then its down to 5 seconds, which seems reasonable. I already have a mock-up of this process so it wouldn't be hard to fully implement. 
2. Only download the cells right around each point. This would be much less data, and would probably be fairly fast, but wouldn't be useful for caching. It would take some work to implement this well - obviously couldn't do a separate query for each point, so would have to send multiple mini bounding boxes, and then return a raster that's mostly empty except for the points? Depending on the implementation, could also end up being slow. 
3. Just make users wait for the data to download - they only need to do it once. 

I would love your thoughts about this.

**Comment by cmahony** [comment]
I like option 1 (cloud downscaling), implemented as a parameter in the downscaling function. so, we default to caching, but the user can opt for cloud downscaling. then the documentation can explain the pros and cons.

**Comment by CeresBarros** [comment]
Yes i like option 1 best as well. But I would say that we can find an "auto" option that is suitable. E.g. if using < X points default to extracting the by point in the server, otherwise extract the full data (or the bouding box). Adding messages at the points where these decisions are made with be important, as well as offering the user the ability to override the option -- e.g. they _might_ want to wait for the full dowload if often changing their points

**Comment by cmahony** [comment]
@kdaust please update this issue for current status

**Comment by kdaust** [comment]
In progress but still needs more work. I have it setup for normal, but need to implement for GCMs. Version 1

**Comment by kdaust** [comment]
I think this is fixed with the new options to pull individual tiles
